packages/storybook/src/utilities/matrix.ts

```diff
@@ -144,6 +151,20 @@ export function createMatrixInteractionsFromStates<
             <p>Focus</p>
             ${createMatrixFromStates(component, states.focus)}
         </div>
+        ${states.visited ? html`
```

Comment:
The `when` directive would be more normal in a fast template. Coupled with the above these would split into separate `when` sections